src/OpenFeature.Extensions.Hosting/Internal/OpenFeatureHostedService.cs

```diff
@@ -0,0 +1,33 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+
+namespace OpenFeature.Internal;
+
+internal sealed class OpenFeatureHostedService(Api api, IEnumerable<FeatureProvider> providers) : IHostedLifecycleService
+{
+    readonly Api _api = Check.NotNull(api);
+    readonly IEnumerable<FeatureProvider> _providers = Check.NotNull(providers);
+
+    async Task IHostedLifecycleService.StartingAsync(CancellationToken cancellationToken)
+    {
```

Comment:
I recommend adding a check for the CancellationToken at strategic points within our loops or before initiating long-running tasks. This would enhance our ability to respond to cancellation requests more gracefully.

```
// Check if cancellation has been requested before proceeding
cancellationToken.ThrowIfCancellationRequested();

```
Also, it might be beneficial to create an issue in OpenFeature to discuss the incorporation of CancellationToken parameters across all asynchronous methods. This would align our practices with best practices for asynchronous .NET APIs and improve the overall reliability and user control of the system.

Reply:
Added cancellation token: https://github.com/open-feature/dotnet-sdk/pull/181/commits/6bd175b58f1b2603165a34c8bfdb68ba93f54525